### PR TITLE
Added support for SPIN (the Standard Projectile ID Nomenclature)

### PR DESCRIPTION
**Pull request by GoldenEpsilon**
This only affects some weapons, but I figure that it's better than nothing. (v2)

### CODE REVIEW

File: defpack tools/Defpack Tools.mod.gml [hunk 1]
```diff
@@ -4431,6 +4458,7 @@ with instance_create(x, y, melee ? CustomSlash : CustomProjectile){
         on_anim = nothing
         on_projectile = sword_proj
         on_grenade = sword_proj
+		is_melee = true
```
Comment (jsburger) [inline defpack tools/Defpack Tools.mod.gml]: Swords are bolt projectiles; they just happen to clear projectiles. I wouldn't consider them a melee attack.

File: defpack tools/Defpack Tools.mod.gml [hunk 2]
```diff
@@ -2943,6 +2952,8 @@ else{
         lightning_refresh()
         hitid = [sprLightningHit,"Lightning Bolt"]
         name = "Lightning Bolt"
+		ammo_type = 5;
+		is_lightning = true;
```
Comment (jsburger) [inline defpack tools/Defpack Tools.mod.gml]: I don't like the blanket coverage provided by "is_lightning." It makes it unclear whether this is an object that can be stretched like lightning or if it just belongs to the "family" of lightning weapons. Considering how lightning bolts are closer to explosions than projectiles I would be careful with this.
That is to say, I don't know how this variable is used, so I don't like it. If it's fine, then it's fine.

**Review by jsburger — CHANGES_REQUESTED** [review]

**Review by GoldenEpsilon — COMMENTED** [review]
Sounds good! Thanks for double checking/catching that

**Comment by GoldenEpsilon** [comment]
now how tf do I make the changes work... does that have to happen on your side, or is it mine?

**Review by jsburger — APPROVED** [review]

**Pull request merged**